src/main/java/decodex/Decodex.java

```diff
@@ -55,46 +54,22 @@ public static void main(String[] args) {
     }
 
     public void run() {
+        ui.printGreeting();
+
         Command command = null;
 
         do {
-            printPromptHeader();
-            String userInput = in.nextLine();
+            String userInput = ui.readInput(in);
             logger.fine("User input: " + userInput);
             try {
                 command = parser.parseCommand(userInput);
-                assert command != null : "Command should not be null";
+                assert command != null : AssertMessages.COMMAND_NOT_NULL;
```

Comment:
For `assert` messages, I feel that there is no need to change the way we have been doing. 

Perhaps there is no need to create constants for this? Similar to logger messages, since they are understood at only one location.